chain/store.go

```diff
@@ -31,25 +32,33 @@ var logStore = logging.Logger("chain.store")
 var headKey = datastore.NewKey("/chain/heaviestTipSet")
 
 type ipldSource struct {
-	// bs is the blockstore from which IPLD objects are pulled
-	bs bstore.Blockstore
+	// cst is a store allowing access
+	// (un)marshalling and interop with go-ipld-hamt.
+	cst *hamt.CborIpldStore
```

Comment:
The tradeoff to using this single store instead of this and the blockstore is that go-filecoin has less control over decoding process.  Practically this means that blocks decoded during load no longer cache their serialized bytes [here](https://github.com/filecoin-project/go-filecoin/blob/master/types/block.go#L127)

Reply:
Check out #3078 for a potential plan on improving the state of this.